Update SEED

paddatrapper opened the pull request:
Notable changes that need addressing:

```
/seed/api/v2/urls.py
    L49-L51

/seed/data_importer/tasks.py
    L771-L773

/seed/serializers/measures.py
    L63-L68

/seed/tasks.py
    L42, L82, L123 - function parameters changed

/seed/templates/seed/_header.html
    L2-L15

/seed/templates/seed/_sidebar.html
    L16-L22

/seed/static/seed/js/controllers/data_quality_admin_controller.js
    L213, L218-L219, L227-L232, L423-L427

/seed/static/seed/js/controllers/data_quality_admin.html
    L105,L167, L172, L177

/seed/static/seed/js/services/data_quality_service.js
    * Moved all API calls to V3 (changes in v2 API need migrating)
    * check parameters in API calls

/seed/static/seed/js/services/columns_service.js
    L47-L70

/seed/static/seed/js/controllers/inventory_list_controller.js
    L440-L442, L450-L452, L556

/seed/static/seed/js/controllers/inventory_detail_controller.js
    L258-L315

/seed/static/seed/js/controllers/data_upload_model_controller.js
    L718-L745, L683-L686

/seed/static/seed/js/controllers/export_inventory_modal_controller.js
    L15-L16, L26-L60
```

macintoshpie committed:
refactor: use xlrd to check for empty xlsx workbook

Removing pandas b/c it takes a long time to install on alpine

macintoshpie committed:
build!: update to django 3.0

Also update other packages which were breaking the app.
NOTE: django-oauth-toolkit update requires additional migrations to be run! You must do this now!

macintoshpie committed:
build: update django-extensions to 3.1.3

Note we had to add explicit ordering to some models due to 3.0 change. See here:
https://github.com/django-extensions/django-extensions/blob/main/CHANGELOG.md#300

macintoshpie committed:
build: remove unused packages and update django-autoslug

Remove django-storages and modeltranslation

macintoshpie committed:
fix: fix wait-for-it by installing coreutils

wait-for-it.sh script was broken. See comment in dockerfile

macintoshpie committed:
build: bump celery to 4.4.7 and unpin billiard

macintoshpie committed:
refactor: remove deprecated 'U' mode for opening files

macintoshpie committed:
test: use assertEqual instead of assertEquals

assertEquals is deprecated

macintoshpie committed:
test: use assertRaisesRegex

assertRaisesRegexp is deprecated

macintoshpie committed:
test: create polyfill for assertDictContainsSubset

assertDictConainsSubset is deprecated

macintoshpie committed:
refactor: use gettext instead of ugettext

ugettext will be removed in django 4.0

macintoshpie committed:
refactor: use importlib instead of imp

imp is deprecated

macintoshpie committed:
test: use COMPRESS_FILTERS for django-compress

COMPRESS_CSS_FILTERS is being deprecated

macintoshpie committed:
build: base requirements cleanup

Removing unused / unnecessarily pinned packages.

macintoshpie committed:
fix: fix django-compressor COMPRESS_FILTERS

macintoshpie committed:
build: update djangorestframework from 3.11 to 3.12

Also update drf-yasg

macintoshpie committed:
style: sate flake8

chore: debug CI failure

chore: debug ci

chore: debug ci

fix(config): fix usage of importlib

chore: bust CI cache

chore: revert CI debugging

macintoshpie committed:
fix: dont use removed package pandas

Commit:
stash

aviveiros11 committed:
Merge pull request #2789 from SEED-platform/2786-fix/sqft-encoding

2786 fix/sqft encoding

axelstudios committed:
feat: add notes modal to inventory list view with code reuse

macintoshpie committed:
refactor(better): add new saved fields

macintoshpie committed:
refactor(better): add r squared to saved result

axelstudios committed:
Merge pull request #2791 from SEED-platform/feat/2511-notes-modal

feat: add notes modal to inventory list view

laurenliz22 committed:
feat(models): update summary tab stylings

Commit:
fix: added support for zip files

Commit:
Merge branch 'develop' into 2693-feature/bsync_support_v2

Commit:
fix: added newly supported schemas

Commit:
fix: removed commented url

macintoshpie committed:
chore: allow CI to downgrade chrome

macintoshpie committed:
chore: allow CI to downgrade chrome

macintoshpie committed:
refactor: always use latest chrome in CI

macintoshpie committed:
chore: remove old flag in CI

aviveiros11 committed:
fix: whitelist better resources in our csp

Commit:
Merge pull request #2798 from SEED-platform/chore/fix-ci-chrome

chore: always use latest chrome in CI

aviveiros11 committed:
fix: only add content-disposition for non html files

Commit:
Merge branch 'develop' into 2693-feature/bsync_support_v2

aviveiros11 committed:
Merge branch 'develop' into fix/content-disposition

axelstudios committed:
fix: added modular csp within server block

aviveiros11 committed:
Merge pull request #2799 from SEED-platform/fix/content-disposition

Fix/content disposition

macintoshpie committed:
Merge branch 'develop' into 2693-feature/bsync_support_v2

macintoshpie committed:
fix: add new bsync schemas to get_schema

refactor!: add migration for renaming default BSync column mapping

macintoshpie committed:
style: sate flake8

macintoshpie committed:
style: sate flake8

Commit:
Merge pull request #2784 from SEED-platform/2693-feature/bsync_support_v2

2693 feature/bsync support v2

Commit:
fix: skipping resource_uses with no type

Commit:
Merge pull request #2800 from SEED-platform/2696-fix/resourceuse_with_no_energyresource_2

fix: skipping resource_uses with no type

macintoshpie committed:
Merge branch 'develop' into refactor/better-saved-results

macintoshpie committed:
Merge pull request #2795 from SEED-platform/refactor/better-saved-results

refactor(better): add new saved fields

macintoshpie committed:
Merge branch 'develop' into feat/analysis-stats

macintoshpie committed:
refactor: tweak summary stats page

macintoshpie committed:
style: sate flake8

Commit:
refactor: moved items around in upload modal

Commit:
Merge branch 'develop' into 1962-refactor/add_icons_to_upload_modal

macintoshpie committed:
Merge pull request #2742 from SEED-platform/feat/analysis-stats

feat(models): add stats to analysis.pys

macintoshpie committed:
fix: correctly check if import file has meters tab

aviveiros11 committed:
feat: add 0.6 placeholder for r^2 value

aviveiros11 committed:
feat: add description to tooltip for a value from 0 to 1

aviveiros11 committed:
feat: make the config more readable

aviveiros11 committed:
feat: new analysis config display

aviveiros11 committed:
feat: more descriptive downloads

Commit:
Merge branch 'develop' into 1962-refactor/add_icons_to_upload_modal

Commit:
refactor: added translations

Commit:
refactor: moved cycle selection and renamed to 'cycle'

aviveiros11 committed:
Merge pull request #2812 from SEED-platform/feat/general-better-integration-tweaks

Feat/general better integration tweaks

macintoshpie committed:
Merge branch 'develop' into fix/check-meters-tab

macintoshpie committed:
refactor: improved handling of BSync skipped meters

macintoshpie committed:
fix(building_file): fix ref to undefined var

fix: invalid syntax

macintoshpie committed:
test: fix meters tab view tests

Commit:
Merge pull request #2815 from SEED-platform/refactor/bsync-skipped-meters

2814 refactor: improved handling of BSync skipped meters

Commit:
Merge branch 'develop' into 1962-refactor/add_icons_to_upload_modal

macintoshpie committed:
fix: handle additional edge cases for BETTER analysis

- only pass meters and readings with 12 readings of values >= 1.0
- dont fail pipeline if one building fails to be created

macintoshpie committed:
style: sate flake8

life’s little, our heads
sad. Redeemed and wasting clay
this chance. Be of use.

Commit:
feat: added initial implementation of new analysis EUI

Commit:
Merge branch 'develop' into 2739-feat/add_eui_analysis

macintoshpie committed:
Merge pull request #2817 from SEED-platform/fix/better-edge-cases

fix: handle additional edge cases for BETTER analysis

aviveiros11 committed:
fix: add helper has child function to the run controller

macintoshpie committed:
fix: correctly filter columns when updating column profile

macintoshpie committed:
Merge pull request #2819 from SEED-platform/2818-fix/bad-display

fix: add helper has child function to the run controller

macintoshpie committed:
Merge branch 'develop' into 2790-fix/inventory-list-col-profile

Commit:
Merge branch 'develop' into 1962-refactor/add_icons_to_upload_modal

Commit:
fix: minor tweaks in layout

Commit:
Merge branch 'develop' into 2739-feat/add_eui_analysis

Commit:
fix: flake

macintoshpie committed:
Merge pull request #2821 from SEED-platform/2790-fix/inventory-list-col-profile

fix: correctly filter columns when updating column profile

macintoshpie committed:
Merge branch 'develop' into fix/check-meters-tab

macintoshpie committed:
Merge branch 'develop' into 1962-refactor/add_icons_to_upload_modal

macintoshpie committed:
Merge pull request #2805 from SEED-platform/1962-refactor/add_icons_to_upload_modal

1962 refactor/add icons to upload modal

Commit:
Merge branch 'develop' into 2739-feat/add_eui_analysis

Commit:
fix: re-organized errors and allows batches with some failures to still complete

Commit:
fix: flake

macintoshpie committed:
Merge branch 'develop' into fix/check-meters-tab

macintoshpie committed:
Merge pull request #2811 from SEED-platform/fix/check-meters-tab

fix: correctly check if import file has meters tab

Commit:
feature: added tests

Commit:
fix: updated tests

Commit:
feat: added tests

Commit:
fix: flake

Commit:
fix: flake

Commit:
Merge branch 'develop' into 2739-feat/add_eui_analysis

aviveiros11 committed:
fix: placeholder changed to actual value input

aviveiros11 committed:
fix: bind the value to the directive

macintoshpie committed:
refactor(bsync): parse meter data from AllResourceTotals

This is required because Audit Template puts some meter data there

macintoshpie committed:
Merge pull request #2829 from SEED-platform/refactor/bsync-AT-meter-import

refactor(bsync): parse meter data from AllResourceTotals

macintoshpie committed:
Merge branch 'develop' into fix/placeholder

macintoshpie committed:
refactor!: link building file meters to property

macintoshpie committed:
Merge pull request #2832 from SEED-platform/refactor/building-meter-link

refactor!: link building file meters to property

Commit:
Merge branch 'develop' into 2739-feat/add_eui_analysis

Commit:
fix: wrong way to access an object

Commit:
feat: added new property_state field to track analyses results

Commit:
Merge pull request #2825 from SEED-platform/2739-feat/add_eui_analysis

2739 feat/add eui analysis

Commit:
Merge branch 'develop' into 2740-feat/property_analyes_results

macintoshpie committed:
Merge branch 'develop' into fix/placeholder

macintoshpie committed:
Merge pull request #2827 from SEED-platform/fix/placeholder

Fix/placeholder

macintoshpie committed:
refactor(better): round resulst to 2 decimal places

macintoshpie committed:
refactor(analyses): add param to auto start analysis

Adds query param to kick things off. Always kick off analyses from the front-end

macintoshpie committed:
style: sate flake8

Commit:
Merge pull request #2845 from SEED-platform/refactor/round-better-results

refactor(better): round result to 2 decimal places

Commit:
Merge branch 'develop' into refactor/auto-start-analysis-param

Commit:
Merge pull request #2848 from SEED-platform/refactor/auto-start-analysis-param

refactor(analyses): add param to auto start analysis

macintoshpie committed:
feat(analyses): auto-poll analysis progress on frontend

Also standardize the progress data keys.
TODO: finish refactoring pipelines other than BETTER to use new pipeline base methods

macintoshpie committed:
style: sate flake8

Commit:
feat: added most recent analysis to inventory detail view

Commit:
Merge branch 'develop' into 2740-feat/property_analyes_results

macintoshpie committed:
refactor(better): use IP units in standalone HTML report

Commit:
fix: removed unneccessary value

Commit:
fix: flake

Commit:
fix: removed unused implementation

nllong committed:
Merge pull request #2855 from SEED-platform/refactor/BETTER-IP-units

refactor(better): use IP units in standalone HTML report

macintoshpie committed:
feat: auto-poll analysis status

Finish frontend polling implementation

macintoshpie committed:
refactor(bsyncr): use updated analysis model

Commit:
fix: saving results to correct property_state

Commit:
fix: added org.id for analysis card link

Commit:
Merge branch 'develop' into 2740-feat/property_analyes_results

Commit:
fix: flake

Commit:
fix: added user_service to inventory_detail so analysis_card would have run_author

macintoshpie committed:
refactor: update bsyncr analysis display

macintoshpie committed:
fix: handle analysis highlights edge cases

Handle cases where results arent stored or are None

macintoshpie committed:
test: fix functional js tests for analysis payload

macintoshpie committed:
test: fix functional js users_payload provider

macintoshpie committed:
test: fix organization provider for detail controller

macintoshpie committed:
style: sate flake8

macintoshpie committed:
refactor: evaluate derived columns in smaller batch size

macintoshpie committed:
refactor: dont run CI on arbitrary pushes

Only run on develop and main push, or any pull request

macintoshpie committed:
Merge pull request #2854 from SEED-platform/2740-feat/property_analyes_results

2740 feat/property analyes results

macintoshpie committed:
refactor(eui): use updated pipeline methods

macintoshpie committed:
refactor: show EUI analysis results on run page

macintoshpie committed:
feat: add analysis polling to analysis details partial

macintoshpie committed:
refactor: auto poll progress for analysis cards

TODO: this is a temporary hack and should be fixed
We should refactor analysis representations so they can manage state themselves

macintoshpie committed:
Merge branch 'develop' into 2858-build/update-ci-triggers

macintoshpie committed:
Merge pull request #2859 from SEED-platform/2858-build/update-ci-triggers

refactor: dont run CI on arbitrary pushes

macintoshpie committed:
Merge branch 'develop' into feat/auto-poll-analyses

nllong committed:
add explicit media path for dev users

Commit:
Merge pull request #2853 from SEED-platform/feat/auto-poll-analyses

feat(analyses): auto-poll analysis progress on frontend

Commit:
feat: added link to analysis run page back to original property

nllong committed:
Merge pull request #2861 from SEED-platform/2842-feat/add-link-to-list

feat: added link to analysis run page back to original property

nllong committed:
Merge remote-tracking branch 'origin/develop' into bug/media-path-in-dev

macintoshpie committed:
refactor(analysis): create method for getting PropertyViews

Also fixes failing test where no PropertyView is found for an AnalysisPropertyView

macintoshpie committed:
fix: add PropertyView IDs to analysis list page

macintoshpie committed:
style: sate flake8

macintoshpie committed:
refactor: dont link to detail view if analysis property DNE

Commit:
Merge pull request #2870 from SEED-platform/fix/analysis-views

Fix/analysis views

macintoshpie committed:
Merge branch 'develop' into build/django-3.0

macintoshpie committed:
fix: import ureg from updated import path

macintoshpie committed:
fix: resolve migration conflict

Commit:
refactor: moved analyses link to sidebar and adjusted navigation of analysis pages

macintoshpie committed:
fix: get analysis highlights from analysis results

macintoshpie committed:
Merge branch 'develop' into bug/media-path-in-dev

Commit:
fix: removed nested config and changed broken min_model_r_squared key

nllong committed:
Merge pull request #2860 from SEED-platform/bug/media-path-in-dev

add explicit media path for dev users

Commit:
Merge branch 'develop' into fix/simplify-config

macintoshpie committed:
fix: handle case where property does not exist

Commit:
Merge branch 'develop' into fix/analysis-highlights

macintoshpie committed:
Merge pull request #2875 from SEED-platform/fix/analysis-highlights

fix: get analysis highlights from analysis results

macintoshpie committed:
fix: initialize label column width

Commit:
Merge pull request #2878 from SEED-platform/fix/label-col-width

fix: initialize label column width

Commit:
Merge branch 'develop' into 2863-refactor/move-analyses-link

Commit:
fix: analyses pages now respond to changing the org

Commit:
Merge branch 'develop' into fix/simplify-config

Commit:
Merge branch 'fix/simplify-config' of https://github.com/SEED-platform/seed into fix/simplify-config

macintoshpie committed:
fix: dq rule serialization

Previously, we tried to use self.instance which didn't yet exist.

Commit:
refactor: storing BETTER setting keys instead of objects

Commit:
refactor: storing BSYNCR setting keys instead of objects

macintoshpie committed:
fix: dq rule serialization

Allow usage of .instance if doing an update of rule

Commit:
fix: updated link

Commit:
Merge pull request #2876 from SEED-platform/fix/simplify-config

fix: removed nested config and changed broken min_model_r_squared key

macintoshpie committed:
test: fix dq rules test

nllong committed:
add recaptcha to docker files

Commit:
Merge branch 'develop' into 2863-refactor/move-analyses-link

macintoshpie committed:
Merge pull request #2874 from SEED-platform/2863-refactor/move-analyses-link

refactor: moved analyses link

macintoshpie committed:
Merge branch 'develop' into fix/rule-serialization

macintoshpie committed:
Merge pull request #2879 from SEED-platform/fix/rule-serialization

fix: dq rule serialization

macintoshpie committed:
Merge branch 'develop' into 2794-fix/derived-col-request

Commit:
Merge pull request #2857 from SEED-platform/2794-fix/derived-col-request

refactor: evaluate derived columns in smaller batch size

nllong committed:
add SITE key to env var

nllong committed:
template tag to parse settings

macintoshpie committed:
refactor: dont add address line 1 to better bsync files

nllong committed:
typo in default site key

nllong committed:
report aws mailer config error

nllong committed:
Merge branch 'develop' into bug/recaptcha-key-in-docker

macintoshpie committed:
refactor: update values saved from BETTER

Save elec and fuel specific fields, conditional on model validity

macintoshpie committed:
style: sate flake8

Commit:
Merge branch 'develop' into build/django-3.0

macintoshpie committed:
refactor: use SEEDs fork of jwt-oauth2

macintoshpie committed:
refactor: add logging and updated user message

macintoshpie committed:
Merge pull request #2736 from SEED-platform/build/django-3.0

build: update to django 3.0

Commit:
Merge branch 'develop' into bug/recaptcha-key-in-docker

macintoshpie committed:
build: bump django to 3.1.13

macintoshpie committed:
fix: dont use deprecated import path

macintoshpie committed:
fix: match new token pattern

Django 3.1 updated the way tokens are generated so it's longer now:
https://docs.djangoproject.com/en/3.2/releases/3.1/#minor-features

macintoshpie committed:
test: avoid too deep recursion

Hypothesis framework was freaking out on some test seeds due to too deep recursion

macintoshpie committed:
fix: add missing migration for EUI analysis service

macintoshpie committed:
test(ci): verify no migrations need to be made

macintoshpie committed:
Merge pull request #2894 from SEED-platform/build/django-3.1

Build/django 3.1

macintoshpie committed:
build: add DEFAULT_AUTO_FIELD setting to supress warnings

Required as of django 3.2

macintoshpie committed:
build: bump django from 3.1 to 3.2

macintoshpie committed:
refactor: use BooleanField(null=True) instead of deprecated field

refactor: add migration

nllong committed:
fix: add default keys to local untracked

nllong committed:
Merge branch 'develop' into bug/recaptcha-key-in-docker

macintoshpie committed:
Merge branch 'develop' into build/django-3.2

macintoshpie committed:
Merge pull request #2897 from SEED-platform/build/django-3.2

Build/django 3.2

Commit:
Merge branch 'develop' into bug/recaptcha-key-in-docker

nllong committed:
Merge pull request #2881 from SEED-platform/bug/recaptcha-key-in-docker

add recaptcha to docker files

macintoshpie committed:
fix: alter seeduser json columns to jsonb type

Django 3.1 broke the usage of "json" type.
See how we fixed this for extra_data: https://github.com/SEED-platform/seed/pull/2407#pullrequestreview-496042761
See more info about 3.1's issue: https://code.djangoproject.com/ticket/31973?cversion=0&cnum_hist=4#comment:4

macintoshpie committed:
test: skip BETTER BSync schema validation

TLDR; we're seeing a strange error from XMLSchema
See here for more info: https://github.com/SEED-platform/seed/pull/2901

nllong committed:
Merge pull request #2901 from SEED-platform/fix/seeduser-json-col-types

fix: alter seeduser json columns to jsonb type

macintoshpie committed:
Merge branch 'develop' into refactor/misc-better-improvements

macintoshpie committed:
refactor: add back BETTER combined savings

Also log warnings for user when models fail

macintoshpie committed:
fix: fix bsyncr analysis creation

macintoshpie committed:
Merge pull request #2887 from SEED-platform/refactor/misc-better-improvements

Refactor/misc better improvements

macintoshpie committed:
fix: fix $scope.org and handle case where analysis progress returns error

Commit:
Merge pull request #2905 from SEED-platform/fix/misc-analyses-bugs

Fix/misc analyses bugs

macintoshpie committed:
refactor: use GH action for stale issues bot

macintoshpie committed:
fix(ci): grant permission to write to issues

macintoshpie committed:
Merge pull request #2906 from SEED-platform/chore/stale-action

refactor: use GH action for stale issues bot

macintoshpie committed:
refactor: use bsync v2.3 for BETTER analysis

nllong committed:
expose better URL as a config setting

nllong committed:
formatting yml

macintoshpie committed:
Merge pull request #2909 from SEED-platform/chore/bump-better-bsync-version

refactor: use bsync v2.3 for BETTER analysis

macintoshpie committed:
Merge branch 'develop' into feat/better-url-setting

macintoshpie committed:
fix: uncomment derived columns org tab

macintoshpie committed:
Merge pull request #2912 from SEED-platform/fix/unhide-derived-cols

fix: uncomment derived columns org tab

macintoshpie committed:
Merge branch 'develop' into feat/better-url-setting

macintoshpie committed:
Merge pull request #2911 from SEED-platform/feat/better-url-setting

Expose better URL as a config setting

macintoshpie committed:
build: prep for 2.12.0 release

macintoshpie committed:
build: bump npm-shrinkwrap.json version

macintoshpie committed:
Merge pull request #2915 from SEED-platform/build/prep-2.12-release

build: prep for 2.12.0 release

paddatrapper committed:
Merge branch 'upstream-develop' into update-develop

paddatrapper committed:
fix build failures

paddatrapper commented:
Looks like the HELIX app doesn't support python 3.9 or needs dependency updates:

```
seed_celery    | Traceback (most recent call last):                                                      
seed_celery    |   File "/usr/bin/celery", line 8, in <module>                      
seed_celery    |     sys.exit(main())                                                                    
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/__main__.py", line 16, in main
seed_celery    |     _main()                                                                             
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/bin/celery.py", line 322, in main
seed_celery    |     cmd.execute_from_commandline(argv)                    
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/bin/celery.py", line 499, in execute_from_commandline
seed_celery    |     super(CeleryCommand, self).execute_from_commandline(argv)))
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/bin/base.py", line 305, in execute_from_commandline                 
seed_celery    |     return self.handle_argv(self.prog_name, argv[1:])        
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/bin/celery.py", line 491, in handle_argv
seed_celery    |     return self.execute(command, argv)                      
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/bin/celery.py", line 415, in execute
seed_celery    |     return cls(                                                                         
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/bin/worker.py", line 223, in run_from_argv
seed_celery    |     return self(*args, **options)                                                                                                                                                                
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/bin/base.py", line 253, in __call__
seed_celery    |     ret = self.run(*args, **kwargs)                                                    
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/bin/worker.py", line 253, in run
seed_celery    |     worker = self.app.Worker(                                                           
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/worker/worker.py", line 99, in __init__                                                                                                          
seed_celery    |     self.app.loader.init_worker()                                                                                                                                                                
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/loaders/base.py", line 114, in init_worker
seed_celery    |     self.import_default_modules()
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/loaders/base.py", line 108, in import_default_modules
seed_celery    |     raise response                                                                      
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/utils/dispatch/signal.py", line 288, in send
seed_celery    |     response = receiver(signal=self, sender=sender, **named)
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/fixups/django.py", line 84, in on_import_modules                                                                                                 
seed_celery    |     self.worker_fixup.validate_models()
seed_celery    |   File "/usr/lib/python3.9/site-packages/celery/fixups/django.py", line 123, in validate_models
seed_celery    |     run_checks()
seed_celery    |   File "/usr/lib/python3.9/site-packages/django/core/checks/registry.py", line 76, in run_checks
seed_celery    |     new_errors = check(app_configs=app_configs, databases=databases)
seed_celery    |   File "/usr/lib/python3.9/site-packages/django/core/checks/urls.py", line 13, in check_url_config
seed_celery    |     return check_resolver(resolver) 
seed_celery    |   File "/usr/lib/python3.9/site-packages/django/core/checks/urls.py", line 23, in check_resolver
seed_celery    |     return check_method()
seed_celery    |   File "/usr/lib/python3.9/site-packages/django/urls/resolvers.py", line 412, in check
seed_celery    |     for pattern in self.url_patterns:
seed_celery    |   File "/usr/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
seed_celery    |     res = instance.__dict__[self.name] = self.func(instance)
seed_celery    |   File "/usr/lib/python3.9/site-packages/django/urls/resolvers.py", line 598, in url_patterns
seed_celery    |     patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
seed_celery    |   File "/usr/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
seed_celery    |     res = instance.__dict__[self.name] = self.func(instance)
seed_celery    |   File "/usr/lib/python3.9/site-packages/django/urls/resolvers.py", line 591, in urlconf_module
seed_celery    |     return import_module(self.urlconf_name)
seed_celery    |   File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
seed_celery    |     return _bootstrap._gcd_import(name[level:], package, level)
seed_celery    |   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
seed_celery    |   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
seed_celery    |   File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
seed_celery    |   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
seed_celery    |   File "<frozen importlib._bootstrap_external>", line 855, in exec_module
seed_celery    |   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
seed_celery    |   File "/seed/config/urls.py", line 14, in <module>
seed_celery    |     from seed.api.base.urls import urlpatterns as api
seed_celery    |   File "/seed/seed/api/base/urls.py", line 11, in <module>
seed_celery    |     from seed.api.v2.urls import urlpatterns as apiv2
seed_celery    |   File "/seed/seed/api/v2/urls.py", line 46, in <module>
seed_celery    |     from helix.views import helix_csv_export, helix_green_addendum, helix_massachusetts_scorecard, helix_dups_export
seed_celery    |   File "/seed/src/helix/helix/views.py", line 36, in <module>
seed_celery    |     from label import label
seed_celery    |   File "/seed/src/label/label/label.py", line 1, in <module>
seed_celery    |     from label.generic_energy_profile import write_generic_energy_profile_pdf
seed_celery    |   File "/seed/src/label/label/generic_energy_profile.py", line 7, in <module>
seed_celery    |     from reportlab.graphics.charts.piecharts import Pie
seed_celery    |   File "/usr/lib/python3.9/site-packages/reportlab/graphics/charts/piecharts.py", line 19, in <module>
seed_celery    |     from reportlab.lib import colors
seed_celery    |   File "/usr/lib/python3.9/site-packages/reportlab/lib/colors.py", line 45, in <module>
seed_celery    |     from reportlab.lib.utils import asNative, isStr
seed_celery    |   File "/usr/lib/python3.9/site-packages/reportlab/lib/utils.py", line 8, in <module>
seed_celery    |     from base64 import decodestring as base64_decodestring, encodestring as base64_encodestring
seed_celery    | ImportError: cannot import name 'decodestring' from 'base64' (/usr/lib/python3.9/base64.py)
```

ClearlyEnergy commented:
> Notable changes that need addressing:
> 
> ```
> /seed/api/v2/urls.py
>     L49-L51
L49: move to v3
L50: move to v3
L51: ours, copy views/v2/pvwatts.py to views/v3/pvwatts.py and update API URL accordingly
> 
> /seed/data_importer/tasks.py
>     L771-L773
keep HELIX version

> 
> /seed/serializers/measures.py
>     L63-L68
> 
keep HELIX version

> /seed/tasks.py
>     L42, L82, L123 - function parameters changed
> 
I think you added the protocol to properly create the invitation URL? 

> /seed/templates/seed/_header.html
>     L2-L15
Use SEED version, we’ll update the logo as needed

> 
> /seed/templates/seed/_sidebar.html
>     L16-L22
Keep line 18 (<i class="fa fa-home"></i>) from HELIX otherwise use SEED
> 
> /seed/static/seed/js/controllers/data_quality_admin_controller.js
>     L213, L218-L219, L227-L232, L423-L427
L213, L218-219: I don’t see a difference
L227-232: use HELIX version
L423-427:  use HELIX version

> 
> /seed/static/seed/js/controllers/data_quality_admin.html
>     L105,L167, L172, L177
I’m not sure you’re referring to the correct file, i.e. I don’t see changes in seed/static/partials/data_quality_admin.html? but you might be looking at a /js file?

> 
> /seed/static/seed/js/services/data_quality_service.js
>     * Moved all API calls to V3 (changes in v2 API need migrating)
>     * check parameters in API calls
> 
> /seed/static/seed/js/services/columns_service.js
>     L47-L70
> 
The methods create_column and create_columns_for_org from the HELIX version should be included.
Make sure that the method delete_column_for_org which seems to be missing from the HELIX version but is in SEED is there

> /seed/static/seed/js/controllers/inventory_list_controller.js
>     L440-L442, L450-L452, L556
> 
L440-442: keep HELIX version but make sure /services/inventory_service.js  get_properties has the matching parameter set, I think the difference is show_sub_org_data.
L450-452: same
L556: same, keep HELIX version but make sure start_data_quality_checks takes in the data_quality_id as parameter

> /seed/static/seed/js/controllers/inventory_detail_controller.js
>     L258-L315
Except for lines 271-273, the rest seems to be duplicates of lines already in the file

> 
> /seed/static/seed/js/controllers/data_upload_model_controller.js
>     L718-L745, L683-L686
> 
L718-745: I don’t see differences, those are lines 650 and onwards in SEED
L683-686: same

> /seed/static/seed/js/controllers/export_inventory_modal_controller.js
>     L15-L16, L26-L60
> ```
L15-16, I think this is only formatting difference, can use SEED version
L26-60 I don’t see differences?

paddatrapper commented:
> seed/urls/v2/urls.py

`/data_quality/` has been renamed to `/data_quality_checks/` in v3. Dropped `/property_measures/` in line with upstream. Moved `/helix_csv_export/`, `/helix_dups_export/`, `/green_addendum/` and `/massachusetts_scorecard/` to v3

paddatrapper committed:
fix merge conflict resolutions

paddatrapper committed:
fix database migrations after seed update
